packages/next/src/server/patch-error-inspect.ts

```diff
@@ -0,0 +1,201 @@
+  let unparsedStack = String(error.stack)
+
+  let idx = unparsedStack.indexOf('react-stack-bottom-frame')
+  if (idx !== -1) {
+    idx = unparsedStack.lastIndexOf('\n', idx)
+  }
+  if (idx !== -1) {
+    // Cut off everything after the bottom frame since it'll be React internals.
+    unparsedStack = unparsedStack.slice(0, idx)
+  }
+
+  const unsourcemappedStack = parseStack(unparsedStack)
+  const sourcemapConsumers = new Map<string, SyncSourceMapConsumer>()
+
+  let sourceMappedStack = ''
+  let sourceFrameDEV: null | string = null
+  for (const frame of unsourcemappedStack) {
+    if (frame.file === null) {
+      sourceMappedStack += '\n' + frameToString(frame)
+    } else if (!shouldIgnoreListByDefault(frame.file)) {
+      const sourcemappedFrame = getSourcemappedFrameIfPossible(
+        frame,
+        sourcemapConsumers
+      )
+
+      if (sourcemappedFrame === null) {
+        sourceMappedStack += '\n' + frameToString(frame)
+      } else {
+        // TODO: Use the first frame that's not ignore-listed
+        if (
+          process.env.NODE_ENV !== 'production' &&
+          sourcemappedFrame.code !== null &&
+          sourceFrameDEV !== null
+        ) {
+          sourceFrameDEV = sourcemappedFrame.code
+        }
+        // TODO: Hide if ignore-listed but consider what happens if every frame is ignore listed.
+        sourceMappedStack += '\n' + frameToString(sourcemappedFrame.stack)
+      }
+    }
+  }
+
+  return (
+    error.message +
+    sourceMappedStack +
+    (sourceFrameDEV !== null ? '\n' + sourceFrameDEV : '')
+  )
+}
+
+export function patchErrorInspect() {
+  Error.prepareStackTrace = prepareUnsourcemappedStackTrace
+
+  // @ts-expect-error -- TODO upstream types
+  // eslint-disable-next-line no-extend-native -- We're not extending but overriding.
+  Error.prototype[inspectSymbol] = function (
+    depth: number,
+    inspectOptions: util.InspectOptions,
+    inspect: typeof util.inspect
+  ): string {
+    // Create a new Error object with the source mapping applied and then use native
+    // Node.js formatting on the result.
+    const newError =
+      this.cause !== undefined
+        ? // Setting an undefined `cause` would print `[cause]: undefined`
+          new Error(this.message, { cause: this.cause })
+        : new Error(this.message)
+
+    // TODO: Ensure `class MyError extends Error {}` prints `MyError` as the name
+    newError.stack = parseAndSourceMap(this)
+
+    const originalCustomInspect = (newError as any)[inspectSymbol]
+    // Prevent infinite recursion.
+    // { customInspect: false } would result in `error.cause` not using our inspect.
+    Object.defineProperty(newError, inspectSymbol, {
+      value: undefined,
+      enumerable: false,
+      writable: true,
+    })
+    try {
+      return inspect(newError, {
+        ...inspectOptions,
+        depth:
+          (inspectOptions.depth ??
+            // Default in Node.js
+            2) - depth,
```

Comment:
I think we wouldn't respect the max depth, if we'd just forward all options. Haven't tested though.